File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentTridentBase.java
package thedarkcolour.futuremc.enchantment;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.inventory.EntityEquipmentSlot;

public class EnchantmentTridentBase extends Enchantment {
    public EnchantmentTridentBase(Rarity rarity, String name) {
        super(rarity, Enchantments.TRIDENT, new EntityEquipmentSlot[] {EntityEquipmentSlot.MAINHAND});
        setRegistryName(name);
        setName(name);
    }
}

File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentTridentBase.kt
package thedarkcolour.futuremc.enchantment

import net.minecraft.enchantment.Enchantment
import net.minecraft.inventory.EntityEquipmentSlot
import thedarkcolour.futuremc.registry.FEnchantments

open class EnchantmentTridentBase(rarity: Rarity, name: String) : Enchantment(
    rarity,
    FEnchantments.TRIDENT,
    arrayOf(EntityEquipmentSlot.MAINHAND)
) {
    init {
        setRegistryName(name)
        setName(name)
    }
}